src/swap/saga.ts

```diff
@@ -247,8 +250,182 @@ export function* swapSubmitSaga(action: PayloadAction<SwapInfo>) {
+  }
+
+  let quoteToTransactionElapsedTimeInMs: number | undefined
+
+  const getTimeMetrics = (): SwapTimeMetrics => ({
+    quoteToUserConfirmsSwapElapsedTimeInMs: swapSubmittedAt - quoteReceivedAt,
+    quoteToTransactionElapsedTimeInMs,
+  })
+
+  try {
+    // Navigate to swap pending screen
+    navigate(Screens.SwapExecuteScreen)
+
+    const wallet = yield* call(getViemWallet, networkConfig.viemChain.celo)
+    if (!wallet.account) {
+      // this should never happen
+      throw new Error('no account found in the wallet')
+    }
+
+    const preparedTransactions = action.payload.quote.preparedTransactions
+    if (preparedTransactions?.type !== 'possible') {
+      // Should never happen
+      throw new Error('No prepared transactions possible')
+    }
+
+    // @ts-ignore typed-redux-saga erases the parameterized types causing error, we can address this separately
+    let nonce: number = yield* call(getTransactionCount, wallet, {
+      address: wallet.account.address,
+      blockTag: 'pending',
+    })
+
+    // Unlock account before executing tx
+    yield* call(unlockAccount, wallet.account.address)
+
+    // Execute transaction(s)
+    yield* put(swapExecute())
+    Logger.debug(TAG, `Starting to swap execute for address: ${wallet.account.address}`)
+
+    const beforeSwapExecutionTimestamp = Date.now()
+    quoteToTransactionElapsedTimeInMs = beforeSwapExecutionTimestamp - quoteReceivedAt
+
+    const txHashes: Hash[] = []
+    for (const preparedTransaction of preparedTransactions.transactions) {
+      const signedTx = yield* call([wallet, 'signTransaction'], {
+        ...preparedTransaction,
+        nonce: nonce++,
+      } as any)
+      const hash = yield* call([wallet, 'sendRawTransaction'], {
+        serializedTransaction: signedTx,
+      })
+      txHashes.push(hash)
+    }
+
+    Logger.debug(TAG, 'Successfully sent swap transaction(s) to the network', txHashes)
+
+    const swapTxHash = txHashes[txHashes.length - 1]
+
+    yield* put(
+      addStandbyTransaction({
+        context: swapExecuteContext,
+        __typename: 'TokenExchangeV3',
+        networkId: networkConfig.defaultNetworkId,
+        type: TokenTransactionTypeV2.SwapTransaction,
+        inAmount: {
+          value: valueToBigNumber(sellAmount)
+            .multipliedBy(guaranteedPrice)
+            .shiftedBy(-toToken.decimals),
+          tokenId: toToken.tokenId,
+        },
+        outAmount: {
+          value: valueToBigNumber(sellAmount).shiftedBy(-fromToken.decimals),
+          tokenId: fromToken.tokenId,
+        },
+        transactionHash: swapTxHash,
+      })
+    )
+
+    const receipt = yield* call([publicClient.celo, 'waitForTransactionReceipt'], {
+      hash: swapTxHash,
+    })
+    Logger.debug('Got swap transaction receipt', receipt)
+    if (receipt.status !== 'success') {
+      throw new Error(`Swap transaction reverted: ${receipt.transactionHash}`)
+    }
+
+    const timeMetrics = getTimeMetrics()
```

Comment:
could we try to use Sentry for time metrics instead of this DIY stuff?

Reply:
Maybe, but there are existing dashboards using this. I'll take a note.
Also I remember we were having difficulty using/analyzing anything metrics related with Sentry.

Reply:
these time metrics aren't used to measure performance. we wanted to be able to analyse swap errors with some data about how long swaps were taking (to answer the question of are users simply just waiting too long to submit the swap after receiving the quote) - it's much easier to do this analysis in Mixpanel if we have all the data there